### PR TITLE
adjustments to facilitate insertion and deletion of items containing

### PR DESCRIPTION
**Pull request by gladiatr72**
```
quotation marks

. _quote_escape(): (new) subs ' for '' for inbound item and returns item
. insert: swap ' for ''' on single-item processing
  run string and list items through _quote_escape() to provide
  proper sqlite3 escaping of single-quote (' -> '')
. delete: swap ' for """ on single-item processing
  ibid: _quote_escape()
```

### CODE REVIEW

**Comment by gladiatr72** [comment]
discovered when trying to insert JSON into a queue

File: salt/queues/sqlite_queue.py [hunk 1]
```diff
@@ -120,6 +121,17 @@ def list_length(queue):
     return len(items)
 
 
+def _quote_escape(item):
+    '''
+    Make sure single quotes are escaped properly in sqlite3 fashion.
+    e.g.: ' becomes ''
+    '''
+
+    rex_sqlquote = re.compile("""'""", re.M)
```
Comment (basepi) [inline salt/queues/sqlite_queue.py]: Do these need the triple double quotes? Doesn't `"''"` suffice?

```
>>> """''""" == "''"
True
```

**Comment by basepi** [comment]
See my comment on the diff. Otherwise it looks good. =)

File: salt/queues/sqlite_queue.py (hunk 1 above)
Comment (gladiatr72) [inline salt/queues/sqlite_queue.py]: Indeed. Thoroughness *does* have its limits! ;)

**Comment by gladiatr72** [comment]
!!! ok.  How the heck did all of those appear... :(

**Comment by basepi** [comment]
Not sure, I think it might be because you merged from develop instead of rebasing or something?

I would recommend resetting this branch to upstream/devleop, and then cherry-picking each of your commits onto the branch. Then you'll have to use `git push -f` to push the branch up. That should fix it.

**Comment by gladiatr72** [comment]
#24053 re-branched and re-pushed.  Evidently my git-fu is weak :P

**Comment by basepi** [comment]
Totally fine, glad you got it figured out! =)

**Pull request closed without merging**